src/app/services/auth.service.ts

```diff
@@ -0,0 +1,132 @@
+   * on the event type, it logs the event and performs necessary actions like 
+   * checking if the user is signed in, setting a refresh token, or clearing the 
+   * user data.
+   * 
+   * @private
+   * @async
+   * @returns {Promise<void>} A promise that resolves when the event handling is complete.
+   */
+  private async listenToAuthEvents() {
+    Hub.listen('auth', async ({ payload }) => {
+      switch (payload.event) {
+        case 'signedIn':
+          this.loggerService.info('User has signed in successfully.');
+          await this.checkIfSignedIn();
+          await this.setRefresh();
+          break;
+        case 'signedOut':
+          this.loggerService.info('User has signed out successfully.');
+          this.user.set(null);
+          this.session.set(null);
+          break;
+        case 'tokenRefresh':
+          this.loggerService.info('Auth tokens have been refreshed.');
+          break;
+        case 'tokenRefresh_failure':
+          this.loggerService.info('Failure while refreshing auth tokens.');
+          break;
+        case 'signInWithRedirect':
+          this.loggerService.info('signInWithRedirect API has successfully been resolved.');
+          break;
+        case 'signInWithRedirect_failure':
+          this.loggerService.info('Failure while trying to resolve signInWithRedirect API.');
+          break;
+      }
+    });
+  }
+
+  /**
+   * Checks if the user is signed in.
+   * 
+   * This method attempts to retrieve the current user and set it to the `user` property.
+   * If the user is signed in, it logs an informational message and a debug message with the user details.
+   * If the user is not signed in, it sets the `user` property to `null`.
+   * 
+   * @returns {Promise<void>} A promise that resolves when the check is complete.
+   * @throws Will throw an error if the user is not signed in.
+   */
+  async checkIfSignedIn() {
+    try {
+      this.user.set(await getCurrentUser());
+      this.loggerService.info('User is signed in.');
+      this.loggerService.debug(JSON.stringify(this.user(), null, 2));
+    } catch (e) {
+      this.user = null;
+      this.loggerService.info('User is not signed in.');
+      this.loggerService.debug(e);
+    }
+  }
+
+  /**
+   * Sets the refresh token and schedules the next refresh based on the token's expiration time.
+   * 
+   * @param {boolean} [forceRefresh=false] - Whether to force a refresh of the authentication session.
+   * @returns {Promise<void>} - A promise that resolves when the refresh operation is complete.
+   * 
+   * @throws {Error} - Throws an error if the token refresh fails.
+   */
+  async setRefresh(forceRefresh = false) {
+    this.session.set(await fetchAuthSession({ forceRefresh: forceRefresh }));
+    this.loggerService.debug(JSON.stringify(this.session(), null, 2));
+    const expiresIn = (this.session().tokens.accessToken.payload.exp * 1000) - Date.now();
```

Comment:
`expiresIn` should be renamed to `refreshInterval = expiresIn/2`. We want to proactively refresh the token before it expires in order to keep the user logged in. Max refresh time can be used to ensure re-auth after a period of time.

Reply:
Do we want /2? Is that too frequent?

Reply:
[9d131c2](https://github.com/bcgov/reserve-rec-public/pull/27/commits/9d131c2fe58b4ea5439f3ffb796fa058313b8b34)